core-java-modules/core-java-streams-5/src/test/java/com/baeldung/aggregateEx/AggregateExceptionHandlerUnitTest.java

```diff
@@ -0,0 +1,180 @@
+package com.baeldung.aggregateEx;
+
+import com.baeldung.aggregateEx.entity.ExceptionAggregator;
+import com.baeldung.aggregateEx.entity.Result;
+
+import static org.junit.Assert.*;
+
+import io.vavr.control.Either;
+import io.vavr.control.Try;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AggregateExceptionHandlerUnitTest {
+
+    private static RuntimeException process(String str) {
+        try {
+            Integer.parseInt(str);
+            return null;
+        } catch (NumberFormatException e) {
+            return new RuntimeException(e);
+        }
+    }
+
+    private static Object transform(String str) {
+        try {
+            return (Integer.parseInt(str));
+        } catch (NumberFormatException e) {
+            return new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void givenExtractedMethod_whenFoundNonInt_thenAggregateException() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        RuntimeException runEx = Arrays.stream(strings)
+                .map(AggregateExceptionHandlerUnitTest::process)
+                .filter(Objects::nonNull)
+                .reduce(new RuntimeException("Errors Occurred"), (o1, o2) -> {
+                    o1.addSuppressed(o2);
+                    return o1;
+                });
+        assertEquals("Errors Occurred", runEx.getMessage());
+        assertEquals(3, runEx.getSuppressed().length);
+    }
+
+    @Test
+    public void givenTryCatchInPipeline_whenFoundNonInts_thenAggregateException() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        RuntimeException runEx = Arrays.stream(strings)
+                .map(str -> {
+                    try {
+                        Integer.parseInt(str);
+                        return null;
+                    } catch (NumberFormatException e) {
+                        return new RuntimeException(e);
+                    }
+                })
+                .filter(Objects::nonNull)
+                .reduce(new RuntimeException("Errors Occurred"), (o1, o2) -> {
+                    o1.addSuppressed(o2);
+                    return o1;
+                });
+        assertEquals("Errors Occurred", runEx.getMessage());
+        assertEquals(3, runEx.getSuppressed().length);
+    }
+
+    @Test
+    public void whenFoundNonInts_thenAggregateExceptionAndReturnOutput() {
```

Comment:
Other similar tests in this class say something like `givenExtractedMethod` - is there a reason this one does not have a "given"?

Reply:
included the when part